core/src/main/java/org/commonjava/maven/ext/core/impl/BaseGroovyManipulator.java

```diff
@@ -158,23 +159,31 @@ void applyGroovyScript( List<Project> projects, Project project, File groovyScri
         {
             try
             {
-                logger.debug ( "Failure when injecting into script {} ", FileUtils.readFileToString( groovyScript, StandardCharsets.UTF_8 ), e );
+                if ( logger.isDebugEnabled() )
+                {
+                    logger.debug( "Failure when injecting into script {}",
+                            FileUtils.readFileToString( groovyScript, StandardCharsets.UTF_8 ), e );
```

Comment:
I wonder if these actually should be `logger.error` given we are in a failure case (which would make the if block redundant)

Reply:
The guard is there in case there is an expensive method call (e.g., not a simple getter) because we pay the price of the method call whether or not the log is enabled (printed).

The case of reading a file seems particularly expensive. Inside a `catch` block may not be as bad since this is an abnormal code path to begin with.

And it may be slightly worse in debug level since it is not enabled by default, but if we also have a quiet mode, then we'd pay the price there as well.

As an aside, I assumed it was intentional, but is the exception `e` passed as an argument there just meant to print `e.toString()` or do we mean to pass the `Throwable` itself as an additional argument (which also shows the stack trace?)

Reply:
I think these should be logger.error (without the guard)

Reply:
I will change it to error level.

Keep in mind that if we remove the guard, then even if we set the logger via `setLevel( Level.OFF )`, the file still gets read, which doesn't seem right. But, I will remove that if you want.

Reply:
I think at that point we need to supply as much information as possible as to what caused the error fir the user to understand. As the build will fail, reading a file doesn't matter in my opinion

Reply:
OK, I've removed the conditional on these and changed it to error level.

I also changed to pass e as an argument (not as a string) as this should provide more information about the error and may be what was intended (not sure), but I think it is better that way as it will have the stack trace.